### PR TITLE
Made the ping data more concise

### PR DESCRIPTION
**Pull request by GateauXD**
- Modified Structure + Component controllers toString to print in the format we want for redis

TODO: Find a way to get translation for entities

### CODE REVIEW

**Comment by GateauXD** [comment]
image
Current Output

File: ComponentController.cs [hunk 1]
```diff
@@ -102,7 +102,8 @@ void OnMouseUp()
     }
     public abstract string GetIcon();
     public string ComponentToString() {
-        return $"    pos = new Vector ({transform.localPosition.x}, {transform.localPosition.y});\n    siz = new Vector ({GetComponent<SpriteRenderer>().size.x}, {GetComponent<SpriteRenderer>().size.y});\n    rot = {gameObject.transform.localEulerAngles.z.ToString("0.00")};";
+        
+        return $"\n{'"' + this.name + '"'}: [{transform.localPosition.x}, {transform.localPosition.y}, {GetComponent<SpriteRenderer>().size.x}, {GetComponent<SpriteRenderer>().size.y}, {gameObject.transform.localEulerAngles.z.ToString("0.00")}],";
```
Comment (Mutilar) [inline ComponentController.cs]: Add GetIcon() before name to determine class via single unicode character.

File: StructureController.cs [hunk 2]
```diff
@@ -452,12 +453,20 @@ public string ToString(string selected)
     }
     public override string ToString()
     {
-        string output = "";
+        string output = $"{{\n\"position\": [{this.center_of_mass.x}, {this.center_of_mass.y}, {this.rotator.rotation.z}],\n\"translation\": [{this.translation.x}, {this.translation.y}],\n";
```
Comment (Mutilar) [inline StructureController.cs]: center_of_mass.x + this.transform.position.x
center_of_mass.y + this.transform.position.y

**Review by Mutilar — CHANGES_REQUESTED** [review]
Added 2 comments

File: ComponentController.cs [hunk 3]
```diff
@@ -102,7 +102,8 @@ void OnMouseUp()
     }
     public abstract string GetIcon();
     public string ComponentToString() {
-        return $"    pos = new Vector ({transform.localPosition.x}, {transform.localPosition.y});\n    siz = new Vector ({GetComponent<SpriteRenderer>().size.x}, {GetComponent<SpriteRenderer>().size.y});\n    rot = {gameObject.transform.localEulerAngles.z.ToString("0.00")};";
+        
+        return $"\n{'"' + this.GetIcon() + this.name + '"'}: [{transform.localPosition.x}, {transform.localPosition.y}, {GetComponent<SpriteRenderer>().size.x}, {GetComponent<SpriteRenderer>().size.y}, {gameObject.transform.localEulerAngles.z.ToString("0.00")}],";
```
Comment (Mutilar) [inline ComponentController.cs]: Let's get rid of the first "\n" of the return. Looks like a duplicate.

File: StructureController.cs [hunk 4]
```diff
@@ -452,12 +453,20 @@ public string ToString(string selected)
...
         {
-            output += component.ToString() + "\n";
+            output += component.ComponentToString() + "\n";
         }
-        return output;
+        output += "]\n\"entities\": [\n";
+
+        foreach (var entity in GameObject.Find("World").GetComponentsInChildren<ProjectileController>())
+        {
+            output += $"\"{entity.name}\": [{entity.transform.position.x}, {entity.transform.position.y},{entity.speed},{entity.transform.localEulerAngles.z.ToString("0.00")}]\n";
```
Comment (Mutilar) [inline StructureController.cs]: Comma at the end of the line?

Comment (Mutilar) [reply StructureController.cs]: ],\n
 ^- since its a list

Comment (GateauXD) [reply StructureController.cs]: image
It's parent is an array also

Comment (Mutilar) [reply StructureController.cs]: I mean for entities, not components?

**Review by Mutilar — APPROVED** [review]

**Pull request merged**